### PR TITLE
feature: Protobuf upgrade

### PR DESCRIPTION
**Pull request by thanethomson**
The primary contribution to #1330.

Must be merged by way of a merge commit (i.e. please do not squash).

Individual PRs have already been reviewed.

:heavy_check_mark: Manual run of the E2E nightlies on the `feature/proto-upgrade` branch: https://github.com/cometbft/cometbft/actions/runs/7025265160

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: .changelog/unreleased/breaking-changes/736-proto-enum-rename.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- `[abci]` Change the proto-derived enum type and constant aliases to the
```
Comment (melekes) [inline .changelog/unreleased/breaking-changes/736-proto-enum-rename.md:1]: it would be nice to have an example here

**Review by melekes — APPROVED** [review]

File: .changelog/unreleased/breaking-changes/736-proto-enum-rename.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  buf-recommended naming conventions adopted in the `abci/v1beta4` proto package
```
Comment (mzabaluev) [inline .changelog/unreleased/breaking-changes/736-proto-enum-rename.md:2]: This line is outdated too, there is no longer `abci/v1beta4`.
I will fix this and above in a small PR.

File: .changelog/unreleased/breaking-changes/736-proto-enum-rename.md (hunk 1 above)
Comment (mzabaluev) [reply .changelog/unreleased/breaking-changes/736-proto-enum-rename.md:1]: Added in #1707

File: .changelog/unreleased/breaking-changes/736-proto-enum-rename.md (hunk 2 above)
Comment (mzabaluev) [reply .changelog/unreleased/breaking-changes/736-proto-enum-rename.md:2]: Fixed in #1707

**Pull request merged**